Amplify/Categories/DataStore/Model/Internal/Schema/AuthRule.swift

```diff
@@ -23,6 +23,14 @@ public enum ModelOperation {
     case read
 }
 
+/// - Warning: Although this has `public` access, it is intended for internal use and should not be used directly
+///   by host applications. The behavior of this may change without warning.
+public enum AuthRuleProvider {
+    case apiKey
+    case oidc
+    case custom(name: String)
```

Comment:
is the plan for codegen to generate strings like `.apiKey`, `oidc`, `.custom("iam")`, and `.custom("userPools")` in the Model+Schema file?

Reply:
The plan is to generate enum cases and/or static method calls like `.apiKey`, `.oidc`, `.iam` `.userPools`.
`AuthRuleProvider` contains only generic providers, and `AWSAuthRuleProvider` adds AWS specific providers via static methods but at the call site won't make any difference

Reply:
would the generate code need `import AWSPluginsCore`?

Reply:
> would the generated code need `import AWSPluginsCore`?

that's a good question... if we create a dependency between the generated models and `AWSPluginsCore` I wonder if the abstraction is worth it, given I don't think it's realistic to drift away from a non-AWS permission model (a security model that doesn't depend on Cognito would require the entire system to change).